src/Umbraco.Deploy.Contrib.Connectors/ValueConnectors/NuPickersValueConnector.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Deploy;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+
+namespace Umbraco.Deploy.Contrib.Connectors.ValueConnectors
+{
+    /// <summary>
+    /// A Deploy connector for the nuPickers property editor, when used with Umbraco content & media.
+    /// </summary>
+    public class NuPickersValueConnector : IValueConnector
+    {
+        private readonly IEntityService _entityService;
+
+        private enum SaveFormat
+        {
+            CSV,
+            JSON,
+            XML,
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NuPickersValueConnector"/> class.
+        /// </summary>
+        /// <param name="entityService">An <see cref="IEntityService"/> implementation.</param>
+        public NuPickersValueConnector(IEntityService entityService)
+        {
+            _entityService = entityService ?? throw new ArgumentNullException(nameof(entityService));
+        }
+
+        public IEnumerable<string> PropertyEditorAliases => Enumerable.Empty<string>();
+
+        /// <summary>
+        /// Gets the deploy property corresponding to a content property.
+        /// </summary>
+        /// <param name="property">The content property.</param>
+        /// <param name="dependencies">The content dependencies.</param>
+        /// <returns>The deploy property value.</returns>
+        public string GetValue(Property property, ICollection<ArtifactDependency> dependencies)
+        {
+            // get the property value
+            var value = property.Value as string;
```

Comment:
True. I had considered it, then then `ParseValue` does a null/whitespace check immediately after - I thought that would cover it?

Question is, if we do a null check before `ParseValue` is called, then do we do an early exit or wrap an `if` around the rest of the logic? _and return what type? `null`, `string.Empty` or `property.Value`?

Reply:
Ahhh yep you're right - it does! 😄
I don't know what makes best sense?... if something is put into this method and isn't a string .. is it okay to just return an empty value (which I guess is what would happen with this logic) ? - or would that be something really incorrect that has happened in this case, so we should throw an exception?

Reply:
Thinking about an exception, but then what if the string value was null, (somehow?)
Then it would be throwing an exception, when there's nothing the user can do about it.

I guess we could initially (before the `string` cast), check if `property.Value` is null, then exit early with a `string.Empty` - sound ok?

Reply:
Actually just looked at the other connectors .. Seems like we're actually just returning `null` if the data isn't what we expect and can handle in some way.
So I guess a check just after casting would accomplish this and then we can exit early - avoid going to the `ParseValue` at all.

Reply:
@clausjensen Done! Thanks.